openedx/tests/xblock_integration/test_review_xblock.py

```diff
@@ -0,0 +1,529 @@
+
+        self.assertIn(expected_h2, response.content)
+
+    @ddt.data(2, 6)
+    def test_review_problems(self, num_desired):
+        """
+        If a user has enough problems to review, they should
+        receive a response where there are review problems for them to try.
+        """
+        self.enroll_student(self.STUDENTS[0]['email'], self.STUDENTS[0]['password'], self.course_actual)
+        self.enroll_student(self.STUDENTS[0]['email'], self.STUDENTS[0]['password'], self.course_review)
+
+        # Loading problems so the learner has enough problems in the CSM
+        self.client.get(reverse(
+            'courseware_section',
+            kwargs={
+                'course_id': self.course_actual.id,
+                'chapter': self.chapter_actual.location.name,
+                'section': self.section1_actual.location.name,
+            }
+        ))
+        self.client.get(reverse(
+            'courseware_section',
+            kwargs={
+                'course_id': self.course_actual.id,
+                'chapter': self.chapter_actual.location.name,
+                'section': self.section2_actual.location.name,
+            }
+        ))
+        self.client.get(reverse(
+            'courseware_section',
+            kwargs={
+                'course_id': self.course_actual.id,
+                'chapter': self.chapter_actual.location.name,
+                'section': self.section3_actual.location.name,
+            }
+        ))
+
+        with self.store.bulk_operations(self.course_actual.id, emit_signals=False):
+            review_section_actual = ItemFactory.create(
+                parent=self.chapter_actual, display_name='Review Subsection'
+            )
+            review_unit_actual = ItemFactory.create(
+                parent=review_section_actual, display_name='Review Unit'
+            )
+
+            review_xblock_actual = ItemFactory.create(  # pylint: disable=unused-variable
+                parent=review_unit_actual,
+                category='review',
+                display_name='Review Tool',
+                num_desired=num_desired
+            )
+
+        # Loading the review section
+        response = self.client.get(reverse(
+            'courseware_section',
+            kwargs={
+                'course_id': self.course_actual.id,
+                'chapter': self.chapter_actual.location.name,
+                'section': review_section_actual.location.name,
+            }
+        ))
+
+        expected_header_text = 'Review Component'
+        # The problems are defaulted to correct upon load
+        # This happens because the problems "raw_possible" field is 0 and the
+        # "raw_earned" field is also 0.
+        expected_correctness_text = 'correct'
+        expected_problems = ['Review Problem 1', 'Review Problem 2', 'Review Problem 3',
+                             'Review Problem 4', 'Review Problem 5', 'Review Problem 6']
+        expected_url_beginning = settings.LMS_ROOT_URL + \
```

Comment:
Since you construct this url beginning two more times below (and a third variant), perhaps you should just construct it once at the class level?